<a href="https://colab.research.google.com/github/Acacia21-code/FlyRankAI-ML-Week1/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Acacia21-code/FlyRankAI-ML-Week1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [1]:
!git clone https://github.com/Acacia21-code/FlyRankAI-ML-Week1.git
%cd FlyRankAI-ML-Week1

fatal: destination path 'FlyRankAI-ML-Week1' already exists and is not an empty directory.
/content/FlyRankAI-ML-Week1


## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I'm going with Lane 4 CTR/Engagement Opportunity Scoring. Some pages rank well in search but still don't get clicked as much as similar-ranking pages do. That's usually a title or meta-description problem not a content problem and it's cheap to fix compared to a full rewrite so it's worth flagging separately.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision: which pages get prioritized for a title/meta-description rewrite
Action: the content team rewrites the snippet for flagged pages
Cost of a wrong call: if you flag the wrong pages, you waste rewrite effort on pages that were never underperforming or you miss genuinely bad pages and leave clicks on the table

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [3]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))

# Only look at pages ranking well
good_position = df[df["position_tier"].isin(["top_3", "page_1"])]
print("Pages ranking top_3 or page_1:", len(good_position))

# Apply a volume floor - low-impression pages give noisy CTR numbers
reliable = good_position[good_position["impressions_90d"] >= 100].copy()
print("Of those, with impressions_90d >= 100:", len(reliable))

# Compare each page's CTR to its tier's average CTR
tier_avg_ctr = reliable.groupby("position_tier")["ctr"].mean()
reliable["tier_avg_ctr"] = reliable["position_tier"].map(tier_avg_ctr)
underperformers = reliable[reliable["ctr"] < reliable["tier_avg_ctr"]]

print("Underperforming candidates:", len(underperformers))
print(tier_avg_ctr)


Total pages: 30000
Pages ranking top_3 or page_1: 14135
Of those, with impressions_90d >= 100: 9166
Underperforming candidates: 6060
position_tier
page_1    0.354760
top_3     0.334128
Name: ctr, dtype: float64


Out of 30,000 total pages, 14,135 rank well enough to matter for this lane (top_3 or page_1). After applying a volume floor of at least 100 impressions in the last 90 days to avoid noisy CTR numbers from low-traffic pages 9,166 pages remain reliable to compare. Of those 6,060 pages have a CTR below their tier's average, meaning about two-thirds of well-ranking pages are underperforming relative to peers at the same position. Average CTR was actually slightly higher for page_1 (35.5%) than top_3 (33.4%) in this slice which is worth flagging rather than assuming top_3 always wins.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This flags candidates, not confirmed problems. Low-impression pages can look like they're underperforming just from random noise the data dictionary itself warns that small-volume tiers can swing CTR by a couple of percentage points from a single click. I'm not claiming these pages definitely need a rewrite, just that they're worth a human second look. This also isn't causal  I don't know why CTR is low, just that it's below the tier norm.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.